src/Database/Expression/QueryExpression.php

```diff
@@ -532,6 +532,20 @@ public function equalFields($left, $right)
         return $this->eq($wrapIdentifier($left), $wrapIdentifier($right));
     }
 
+    /**
+     * Adds a new type convertion to the expression object in the form "CAST(field AS type)".
+     *
+     * @param string $field Database field to be cast against type
+     * @param string $type target data type to cast field to it
+     * @return $this
+     */
+    public function convert($field, $type)
+    {
+        $this->add("CAST($field AS $type)");
```

Comment:
I have rather conflicted feelings about this. Building the SQL immediately basically turns the cast into a non-expression, as it cannot be traversed, transformed, etc., and it's different to what all other expression builder methods do. Furthermore this neither supports automatic identifier quoting, nor does it allow to pass in expressions for the field, which is different to other expressions too.

In my own projects I've used dedicated function expression classes for casts, as they can benefit from having return types applied.

Reply:
Yup, this looks pretty unsafe.